File: app/java/src/main/java/com/example/production/application/port/in/command/StockStatusChangeCommand.java
package com.example.production.application.port.in.command;

import com.example.production.domain.model.inventory.StockStatus;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

/**
 * 在庫状態変更コマンド
 */
@Data
@Builder
public class StockStatusChangeCommand {
    private String locationCode;
    private String itemCode;
    private BigDecimal quantity;
    private StockStatus fromStatus;
    private StockStatus toStatus;
}

File: app/csharp/src/ProductionManagement.Application/Port/In/Command/StockStatusChangeCommand.cs
using ProductionManagement.Domain.Models.Inventory;

namespace ProductionManagement.Application.Port.In.Command;

/// <summary>
/// 在庫状態変更コマンド
/// </summary>
public class StockStatusChangeCommand
{
    public required string LocationCode { get; init; }
    public required string ItemCode { get; init; }
    public required decimal Quantity { get; init; }
    public required StockStatus FromStatus { get; init; }
    public required StockStatus ToStatus { get; init; }
}
